homeassistant/components/fritz/common.py

```diff
@@ -0,0 +1,266 @@
+
+        self._devices: Dict[str, Any] = {}
+        self.scan_devices()
+
+        async_track_time_interval(
+            self.hass, self.scan_devices, timedelta(seconds=TRACKER_SCAN_INTERVAL)
+        )
+
+    def is_ok(self):
+        """Return status."""
+        return self.success, self.error
+
+    @property
+    def unique_id(self):
+        """Return unique id."""
+        return self._unique_id
+
+    @property
+    def fritzbox_model(self):
+        """Return model."""
+        return self._device_info["model"].replace("FRITZ!Box ", "")
+
+    @property
+    def device_info(self):
+        """Return device info."""
+        return self._device_info
+
+    @property
+    def devices(self) -> Dict[str, Any]:
+        """Return devices."""
+        return self._devices
+
+    @property
+    def signal_device_new(self) -> str:
+        """Event specific per Fritzbox entry to signal new device."""
+        return f"{DOMAIN}-device-new"
+
+    @property
+    def signal_device_update(self) -> str:
+        """Event specific per Fritzbox entry to signal updates in devices."""
+        return f"{DOMAIN}-device-update"
+
+    def _update_info(self):
+        """Retrieve latest information from the FRITZ!Box."""
+        if not self.success:
+            return None
+
+        return self.fritzhosts.get_hosts_info()
+
+    def scan_devices(self, now: Optional[datetime] = None) -> None:
+        """Scan for new devices and return a list of found device ids."""
+
+        _LOGGER.debug("Checking devices for Fritz!Box router %s", self.host)
+
+        new_device = False
+        for known_host in self._update_info():
+            if not known_host.get("mac"):
+                continue
+
+            dev_mac = known_host["mac"]
+            dev_name = known_host["name"]
+            dev_ip = known_host["ip"]
+            dev_home = known_host["status"]
+
+            dev_info = Device(dev_mac, dev_ip, dev_name)
+
+            if dev_mac in self._devices:
+                self._devices[dev_mac].update(dev_info, dev_home)
+            else:
+                device = FritzScannerEntity(dev_mac)
+                device.update(dev_info, dev_home)
+                self._devices[dev_mac] = device
+                new_device = True
+
+        async_dispatcher_send(self.hass, self.signal_device_update)
+        if new_device:
+            async_dispatcher_send(self.hass, self.signal_device_new)
+
+    def _fetch_device_info(self):
+        """Fetch device info."""
+        info = self.connection.call_action("DeviceInfo:1", "GetInfo")
+        return {
+            "identifiers": {
+                # Serial numbers are unique identifiers within a specific domain
+                (DOMAIN, self.unique_id)
+            },
+            "name": info.get("NewModelName"),
+            "manufacturer": "AVM",
+            "model": info.get("NewModelName"),
+            "sw_version": info.get("NewSoftwareVersion"),
+        }
+
+
+class FritzScannerEntity:
```

Comment:
It looks like these aren't Home Assistant entities so it would make sense to call it something different maybe `FritzDevice`

Reply:
Done